pkg/protocol/http1/server.go

```diff
@@ -413,6 +413,11 @@ func writeErrorResponse(zw network.Writer, ctx *app.RequestContext, serverName [
 }
 
 func writeResponse(ctx *app.RequestContext, w network.Writer) error {
+	// Skip default response writing logic if it has been hijacked
+	if ctx.Response.GetHijackWriter() != nil {
+		return ctx.Response.GetHijackWriter().Finalize()
```

Comment:
If the user also calls `Finalize`, it seems it will cause problem. How to avoid this happen?

Reply:
Good one, involving another interface will work, but will bring some costs by the assertion.
I will bench it to see the diff. If the overhead is too high, it can only be explained in the comments.